browser/brave_shields/ad_block_service_browsertest.cc

```diff
@@ -1790,6 +1798,54 @@ IN_PROC_BROWSER_TEST_F(CosmeticFilteringFlagDisabledTest,
   ASSERT_EQ(true, EvalJs(contents, "checkSelector('.ad', 'display', 'block')"));
 }
 
+#if BUILDFLAG(ENABLE_PLAYLIST)
+
+namespace playlist {
+
+class CosmeticFilteringPlaylistFlagEnabledTest : public AdBlockServiceTest {
+ public:
+  CosmeticFilteringPlaylistFlagEnabledTest() {
+    feature_list_.InitAndEnableFeature(features::kPlaylist);
+  }
+
+ private:
+  base::test::ScopedFeatureList feature_list_;
+};
+
+// Ensure cosmetic filtering occurs always when AllowCosmeticFiltering() is
+// called.
+IN_PROC_BROWSER_TEST_F(CosmeticFilteringPlaylistFlagEnabledTest,
+                       AllowCosmeticFiltering) {
+  ASSERT_TRUE(InstallDefaultAdBlockExtension());
+  const GURL url =
+      embedded_test_server()->GetURL("b.com", "/cosmetic_filtering.html");
+
+  // Set cosmetic filtering is not enabled.
+  brave_shields::SetCosmeticFilteringControlType(
+      content_settings(), brave_shields::ControlType::ALLOW, url);
+  UpdateAdBlockInstanceWithRules("b.com###ad-banner");
+
+  auto* playlist_service_factory = PlaylistServiceFactory::GetInstance();
+  auto* playlist_service =
+      playlist_service_factory->GetForBrowserContext(browser()->profile());
+  auto* download_request_manager =
+      playlist_service->download_request_manager_.get();
```

Comment:
Could you please remove access to private fields?
`playlist_service->download_request_manager_` and `download_request_manager->web_contents_`

If we need to get webContents from playlist_service, better to add and implement `GetWebContentsForTesting()` method.

Tests should check a public interface and shouldn't know about an implementation => better to remove all the mentions of `download_request_manager` here.

Reply:
Done.